lib/ddtrace/transport/http.rb

```diff
@@ -75,13 +76,27 @@ def default_headers
       end
 
       def default_hostname
+        return default_url.hostname if default_url
+
         ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_HOST, Datadog::Ext::Transport::HTTP::DEFAULT_HOST)
       end
 
       def default_port
+        return default_url.port if default_url
+
         ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_PORT, Datadog::Ext::Transport::HTTP::DEFAULT_PORT).to_i
       end
 
+      def default_url
```

Comment:
what we've done in other languages is use `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` to build the url, that way once the app loads/is configured, then we only have a url we are working with.

https://github.com/DataDog/dd-trace-py/blob/2dff53f643f71fc4b9f414a5d85961ab130d99a1/ddtrace/tracer.py#L74-L79

Not sure exactly how that translates here, since I am guessing `default_hostname`/`default_port` are what are used to create the connection?

Reply:
I think a bit lower down `hostname` and `port` get coerced into a url, but it's a bit tricky since we're resolving the priority of `host / port in code` > `url env var` > `host/port env vars` here, before the url gets formed. which is why there's this roundabout "extract the hostname/port from the url being set via env var", to avoid having to re-structure the other parts of transport too much. @marcotc may have a cleaner solution/idea here.

Reply:
I think this is ok for now, but I agree that supporting only `url` input is a better way forward.
It requires more changes to the transport that it would be comfortable for this PR, though.